### DISCUSSION TITLE
BAP - dealing with clouds and distance to other clouds?

### DISCUSSION THREAD

**Discussion by Antguz** [davidfrantz/force#328] [General]
Hi community,

I am processing Best Available Pixel (BAP) data from lowland tropical forests, where clouds are always present. Thus, I am looking for guidance on this issue.

First, I have noticed that many scenes ingested in Level 2 ARD do not seem to detect certain types of cumulus clouds. For example, take a look at these OVV images derived from Landsat 8:

![20190929_LEVEL2_LND08_OVV](https://github.com/davidfrantz/force/assets/7254767/1289b304-61fd-49a1-a4d6-755f0a6a9717) 
![20160531_LEVEL2_LND08_OVV](https://github.com/davidfrantz/force/assets/7254767/641ea8bd-5f5b-4039-9ba6-de8eb33ef24d)

How can I adjust the settings to make these types of clouds more detectable? (or can be detectable?) I do not want to modify the Fmaks thresholds without discussing it first. Currently, I am using the default values.

Assuming that cloud detection is not an issue, I have noticed when calculating the BAP for a given year that the distance-to-clouds score does not seem to be applied to all types of clouds. For instance, look at these Level 3 BAP and distance score scenes where all the pixels were selected from the same Level 2 image:

![Screenshot from 2024-06-28 21-53-33](https://github.com/davidfrantz/force/assets/7254767/f4550812-8ffc-4514-bb18-a94c4fc6636f)
![Screenshot from 2024-06-28 21-53-54](https://github.com/davidfrantz/force/assets/7254767/aa9eefa5-3fff-465b-8f2d-7ca11d7f8a1b)

It appears that there are some gaps in the data due to clouds that do not have a distance score. The documentation says distance to clouds and shadows, but similar to #224 the score may need to be applied to all the clouds. 

Can I get any guide? 

Thanks for you time.

AG.

**Comment by davidfrantz** [comment]
Hi AG,

this doesn't look very good in this case. Where on Earth are we? Can you send one of those scene IDs?
I am a bit curious as to why this happens...

The cloud distance issue is because the clouds were not detected.
If it cannot be solved at the level of the cloud mask, you could try using the haze score, and/or the correlation score.

Best,
David

**Reply by Antguz** [reply]
Hi David,

This is the Caribbean lowland forest across Costa Rica. 

In this [link](https://drive.google.com/file/d/10QTYeqg1y16ws8Wxx0-7fTRMs_ueT5g8/view?usp=sharing), you will find some scenes from Landsat at different years. 

Even using a lower Fmask threshold (0.125) clouds does not seems to be detected. Also it seems to me this is more frequent in Landsat 8.

Let me know what I can do ...

Regards.

AG.

**Comment by davidfrantz** [comment]
Hi AG.

so... the issue is FMask's strong reliance on the thermal information. In this area, the clouds appear to be very warm. There is less than 1°C temperature difference between the land and the omitted cumuli. The temperature distribution of the land also overlaps that of those clouds.

This is quite an issue here and not easy to resolve...

I tested with this image: LC08_L1TP_015053_20200204_20200823_02_T1 L2

This is the quicklook of the masks:

![20200204_LEVEL2_LND08_OVV](https://github.com/user-attachments/assets/500b0c97-f776-4d77-b5ff-d11a02c96740)

This is the mask produced by USGS (downloaded from EarthExplorer). It is probably a bit better, but there is still quite some underdetection:

![LC08_L1TP_015053_20200204_20200823_02_T1_QB](https://github.com/user-attachments/assets/9872b990-d03e-4025-a83f-ebbf005e2c1b)

If you remove the margin of the temperature probability entirely (``CLOUD_THRESHOLD  = 0``), it becomes much better:

![20200204_LEVEL2_LND08_OVV](https://github.com/user-attachments/assets/55077fb3-7b05-4b67-b13a-56e87f64b10e)

This could be an option here, although I would not recommend it as the future default. But in this particular case, it seems to help...

Best, 
David

**Reply by Antguz** [reply]
Thanks David for the analysis. 

When you said that `CLOUD_THRESHOLD  = 0` could be an option here you are recomending that for these images with this underdetection or for all the scenes that comes from this site? No sure how appropriate is to use different thresholds for the same project.

Again thanks for your guide.

AG.

**Comment by davidfrantz** [comment]
Hi AG,

I would try with one threshold first and see what happens. If you find some season-dependent results (or weather-dependent?), you could also justify different thresholds, I guess.

Best,
David
